packages/http-client-csharp/generator/Microsoft.Generator.CSharp.Input/src/InputTypes/InputDecoratorInfo.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Generator.CSharp.Input
+{
+    public class InputDecoratorInfo
+    {
+        public InputDecoratorInfo(string name, IReadOnlyDictionary<string, object> arguments)
+        {
+            Name = name;
+            Arguments = arguments;
+        }
+        public string Name { get; }
+        public IReadOnlyDictionary<string, object> Arguments { get; }
```

Comment:
would it be better if this is `BinaryData` instead of `object`? This might avoid boxing

Reply:
The type of argument value is any, it can be string/int/float. We use the `object` to reference this, and will serialize to the right value. If we use `BinaryData`, we may lost the type information of the value, and cannot correct serialize it

Reply:
BinaryData stores the raw data from the json, and we also will know which value this is when we use it. This is not an issue.
Also in our generated code, `Record<unknown>` always generates as `IDictionary<string, BinaryData>` or its readonly variant.

Reply:
make sense. I will update it. thanks